src/LineageTree/tree_styles.py

```diff
@@ -8,12 +8,12 @@
 
 class abstract_trees(ABC):
     def __init__(
-        self, lT: lineageTree, root: int, node_length: int, end_time: int
+        self, lT: lineageTree, root: int, downsample: int, end_time: int
     ):
         self.lT = lT
         self.root = root
-        self.node_length = node_length
-        self.end_time = end_time
+        self.downsample = downsample
+        self.end_time = end_time if end_time else self.lT.t_e
```

Comment:
`end_time` is a mandatory parameter here, shouldn't it be by default `None` to make sure that it can not be given?